src/pages/groupComparison/vennDiagram/VennUtils.spec.ts

```diff
@@ -0,0 +1,240 @@
+import {assert} from "chai";
+import {getApproximateRegionArea, getRegionArea} from "./VennUtils";
+
+describe("VennUtils", () => {
+    describe("getRegionArea", ()=>{
+        it("one rectangle", ()=>{
+            assert.approximately(getRegionArea(
+                ["rect1"],
+                { rect1: {x:0, y:0, xLength:5, yLength:10}}
+            ), 50, 0.01);
+        });
+        it("two disjoint rectangles", ()=>{
+            const rects = {
+                rect1: {x:0, y:0, xLength:5, yLength:10},
+                rect2: {x:10, y:10, xLength:10, yLength:20}
+            };
+            assert.approximately(getRegionArea(
+                ["rect1", "rect2"],
+                rects
+            ), 0, 0.01);
+            assert.approximately(getRegionArea(
+                ["rect1"],
+                rects
+            ), 50, 0.01);
+
+            assert.approximately(getRegionArea(
+                ["rect2"],
+                rects
+            ), 200, 0.01);
+        });
+        it("rectangle containing another rectangle", ()=>{
+            const rects = {
+                rect1: {x:0, y:0, xLength:5, yLength:10},
+                rect2: {x:1, y:1, xLength:2, yLength:4}
+            };
+            assert.approximately(getRegionArea(
+                ["rect1", "rect2"],
+                rects
+            ), 8, 0.01);
+            assert.approximately(getRegionArea(
+                ["rect1"],
+                rects
+            ), 42, 0.1);
+
+            assert.approximately(getRegionArea(
+                ["rect2"],
+                rects
+            ), 0, 0.01);
+        });
+        describe.skip("random testing with 2 and 3 rectangles", ()=>{
```

Comment:
skip?

Reply:
these are very costly (several seconds) tests doing [Monte Carlo integration](https://en.wikipedia.org/wiki/Monte_Carlo_integration). I was only using them while testing to test the area calculations. I don't think it's worth running normally